kanidmd/src/lib/value.rs

```diff
@@ -30,11 +30,11 @@ lazy_static! {
     };
     static ref INAME_RE: Regex = {
         #[allow(clippy::expect_used)]
-        Regex::new("^((\\.|_).*|.*(\\s|@|,|/|\\\\|=).*|\\d+|root|nobody|nogroup|wheel|sshd|shadow|systemd.*)$").expect("Invalid Iname regex found")
+        Regex::new("^((\\.|_).*|.*(\\s|:|;|@|,|/|\\\\|=).*|\\d+|root|nobody|nogroup|wheel|sshd|shadow|systemd.*)$").expect("Invalid Iname regex found")
```

Comment:
How does Oauth2 prevent `;` and `:` from being used here?

Reply:
Basic auth for the oauth2 resource server requires "name:secret". If the name contains a : this would confuse our ability to check this, so we remove : from being in names to at least remove a possible behaviour where someone could shoot themself in the foot. Remember, this is the name type (ie uid or login name) not displayname, which has no limits. ; is similarly removed because it can confuse other things.